### DISCUSSION TITLE
Discussion of Patient Letter Generation Research

### DISCUSSION THREAD

**Discussion by amaiaita** [nhsengland/swpc_synthea#20] [General]
**our purpose**
- creating patient letters that are linked to a patients record 
- can take a range of inputs so it can be generalised 

**avenues to investigate**
- previous body's of work that generate patient letters (internal + external)
- patient letter templates for hypertension related stuff

**Comment by amaiaita** [comment]
# **info from @xiyaozhuang  on the work in PrivFP:**

the free text in created using GenAI, with models downloaded and inferred using Ollama. calls to the LLM are made via LangChain. models not provided on Ollama can be downloaded elsewhere and quantised with llama.cpp for low resource compute. Synthea International is used to create synthetic json patient data for the LLM to ingest.
 
some background exploration: https://nhsengland.github.io/privfp-experiments/open-source-llm-exploration/
our source code: https://github.com/nhsengland/privfp-experiments/blob/main/src/generate/llm.py
 
Synthea: https://github.com/synthetichealth/synthea-international
Ollama: https://ollama.com/
LangChain: https://python.langchain.com/v0.2/docs/introduction/
llama.cpp: https://github.com/ggerganov/llama.cpp

# **insight from @scarlett-k-nhs **:
So the repo I found was named entity extraction related - although I thought I'd have a quick look, and this looks like quite a good resource (https://github.com/AI-in-Health/MedLLMsPracticalGuide) - it's all about medicial related LLMs
And at the bottom there is a whole list of papers
- This is a fantastic resource for all sorts of usecases of LLMs for medical purposes, incredibly well structured + regularly updated

**Comment by amaiaita** [comment]
GatorTronGPT 
https://arxiv.org/pdf/2203.03540

- Trained specifically for clinical text generation 
- However - not UK focused. 
- Generally does medical text, not specific to patient letters - so wouldn’t necessarily follow the structure we need
- How would we incorporate it into the right template/format
- Focuses more on producing insight from medical text rather than producing text 
- What its been tested on
  - Can do NER specific to medical semantic categories (e.g. diseases, medications etc)
  - Semantic textual similarity
  - Natural language inference, filling the gaps in sentences correctly 
- Answering medical questions - this is most intreiguing, could this be used to generate patient letters if the question asked is correct.

**Comment by amaiaita** [comment]
Medalpaca 
https://huggingface.co/medalpaca/medalpaca-7b
	- Aims to improve medical dialogue tasks
	- Is at the level of medical students rather than medical professionals - potential limitation
	- Could be used to model the question into patient letter generation, however how do we format the inputs from the record to be included in the question?
Very limited insight in their documentation

**Comment by amaiaita** [comment]
Discussion of privacy fingerprint usage of LLM for creating clinical notes with @willpoulett 
https://github.com/nhsengland/privfp-experiments/blob/main/src/generate/llm.py

- uses json synthea output and feeds it directly into the system prompt, but could be adapted to use f strings to only use some of the outputs. 
- currently use llama 3 and it works quite well 
- this file could be useful to use: https://github.com/nhsengland/privfp-experiments/blob/main/notebooks/full_pipeline_example.ipynb?short_path=f52c1fe 
- we could use a combination of their pipeline, an altered system prompt to be specific to clinical letters, and a medically trained LLM instead of llama
- theyve struggled with the fact that it doesnt always include all the given data - we'd need to be careful with that 
- significant speed limitations for big models - how would we circumvent this?

**Comment by amaiaita** [comment]
Analysis of GPT-4V for medical usecases
https://arxiv.org/pdf/2310.09909
	
![image](https://github.com/user-attachments/assets/bb996f4a-cdf4-40ee-96a1-f458e14933d1)

	- Often hallucinates. But is this just for stuff coming from images rather than text data? 
![image](https://github.com/user-attachments/assets/98bcb70b-f93c-4ad3-8b0a-2df87e5744e0)

	
	- Pretty good with a very minimal prompt. 
	- Potentially helpful for the other way around 
This is incredibly cool that it's able to do this.

**Comment by amaiaita** [comment]
ChatGPT: the future of discharge summaries?
https://www.thelancet.com/journals/landig/article/PIIS2589-7500(23)00021-3/fulltext
Seems like exactly what we want to do

**Comment by amaiaita** [comment]
Patient letter examples:
https://www.gadesurgery.co.uk/wp-content/uploads/2021/02/DIABETES-HYPERTENSION-RECALL-WEBSITE-LETTER.pdf
https://www.gadesurgery.co.uk/wp-content/uploads/2021/02/WEB-Home-BP-Letter.pdf
[Template-Letter-for-GP-NMS-Referral-for-Patients-Newly-Prescribed-Hypertension-Medicine.docx](https://github.com/user-attachments/files/16783592/Template-Letter-for-GP-NMS-Referral-for-Patients-Newly-Prescribed-Hypertension-Medicine.docx)

Some good templates: https://www.westyorkshireandharrogatehealthyhearts.co.uk/resources-to-use-with-patients
Post natal discharge letter for women with hypertension in pregnancy: https://rightdecisions.scot.nhs.uk/m/kkcnkcpt/pn-htn-v5-discharge-letter.pdf

**Comment by benwallace2nhs** [comment]
**Letters** 

Outpatient letter standard from PRSB 

https://theprsb.org/wp-content/uploads/2018/02/Outpatient-Letter-Examples.pdf

NHS England Letter template 

https://www.england.nhs.uk/wp-content/uploads/2023/02/B2012iii-patient-letter-template-appointment-confirmation.docx

Information about clinical letters from ROH

https://roh.nhs.uk/services-information/outpatients/clinical-letters

and digital letters https://roh.nhs.uk/services-information/outpatients/drdoctor

&nbsp;

NHS Appointment letter template 

https://www.pdffiller.com/jsfiller-desk13/?mode=force_choice&requestHash=81f26f99cb0ba3deeaf388061d75b5215a5f98c7aa3b9a8e198de4a9644b0c18&lang=en&projectId=1610819746&PAGE_REARRANGE_V2_MVP=true&richTextFormatting=true&isPageRearrangeV2MVP=true&jsf-page-rearrange-v2=true&jsf-redesign-full=true&act-notary-pro-integration=false&jsf-fake-edit-embedded=true&isSkipEditorLoadFrequency=true&jsf-desktop-ux-for-tablets=false&jsf-probability-70=true&jsf-socket-io=false&routeId=5b3e18f2c53f18a20ad6fd6ff17edf30#9e57f6480b2f4df398463d06387f78ef

&nbsp;

**Discharge summary template** 

**NHS Digital** 

https://data.developer.nhs.uk/specifications/NHS-CDA-eDischarge/Examples/TransferOfCare/POCD_EX000026GB01_01.html

alternate

&nbsp;https://data.developer.nhs.uk/specifications/NHS-CDA-eDischarge/Examples/TransferOfCare/POCD_EX000026GB01_02.html

Royal united hospital bath 

https://www.ruh.nhs.uk/Training/Prospectus/Clinical_systems/Millennium/documents/training_manuals/Doctors/Discharge_Summary_Eguide.pdf

&nbsp;

**Clinical Note**

**Care Patron template https://www.carepatron.com/templates/clinical-note-template**
